lazy_theta_star_p_planner/include/lazy_theta_star_p_planner/lazy_theta_star_p_planner.hpp

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 Anshumaan Singh
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef LAZY_THETA_STAR_P_PLANNER__LAZY_THETA_STAR_P_PLANNER_HPP_
+#define LAZY_THETA_STAR_P_PLANNER__LAZY_THETA_STAR_P_PLANNER_HPP_
+
+#include <iostream>
+#include <cmath>
+#include <string>
+#include <chrono>
+#include <queue>
+#include <algorithm>
+#include <memory>
+#include <vector>
+#include "rclcpp/rclcpp.hpp"
+#include "nav2_core/global_planner.hpp"
+#include "nav_msgs/msg/path.hpp"
+#include "nav2_util/robot_utils.hpp"
+#include "nav2_util/lifecycle_node.hpp"
+#include "nav2_costmap_2d/costmap_2d_ros.hpp"
+#include "nav2_costmap_2d/cost_values.hpp"
+#include "nav2_util/node_utils.hpp"
+#include "lazy_theta_star_p_planner/lazy_theta_star_p.hpp"
+
+namespace lazyThetaStarP_Planner
+{
+
+class LazyThetaStarP_Planner : public nav2_core::GlobalPlanner
+{
+public:
+  std::shared_ptr<tf2_ros::Buffer> tf_;
+  nav2_util::LifecycleNode::SharedPtr node_;
```

Comment:
As well here, this will actually be a weak pointer when you pull in the changes.

Also move all the member variables to the protected domain, we shouldn't have any of that public